test/helpers/cilium.go

```diff
@@ -333,6 +349,96 @@ func (c *Cilium) ReportFailed(commands ...string) {
 		fmt.Fprint(wr, res.Output())
 	}
 	fmt.Fprint(wr, "StackTrace Ends\n")
+	c.ReportFailedFull()
+}
+
+//ReportFailedFull runs a variety of commands and write the results to
+//testResultsPath
+func (c *Cilium) ReportFailedFull() {
+	reportCommands := map[string]string{
+		"sudo cilium endpoint list -o json": "endpoint_list_txt",
+		"sudo cilium service list -o json":  "service_list.txt",
+		"sudo cilium config":                "config.txt",
+		"sudo cilium bpf lb list":           "bpf_lb_list.txt",
+		"sudo cilium bpf ct list global":    "bpf_ct_list.txt",
+		"sudo cilium bpf tunnel list":       "bpf_tunnel_list.txt",
+		"sudo cilium policy get":            "policy_get.txt",
+		"sudo cilium status":                "status.txt",
+	}
+
+	reportEndpointsCommands := map[string]string{
+		"cilium endpoint get %s":    "endpoint_%s.txt",
+		"cilium bpf policy list %s": "bpf_policy_list_%s.txt",
+	}
+
+	testPath, err := ReportDirectory()
+	if err != nil {
+		c.logCxt.Errorf("Can't create Test results path '%s' err '%s'", testPath, err)
+		return
+	}
+	reportMap(testPath, reportCommands, c.Node)
+
+	//Endpoint specific information:
+	eps, err := c.GetEndpointsIds()
+	if err != nil {
+		c.logCxt.Errorf("Can not get endpoint ids")
+	}
+	for _, ep := range eps {
+		for cmd, logfile := range reportEndpointsCommands {
+			command := fmt.Sprintf(cmd, ep)
+			res := c.Node.Exec(fmt.Sprintf("sudo %s", cmd))
+
+			err = ioutil.WriteFile(
+				fmt.Sprintf("%s/%s", testPath, fmt.Sprintf(logfile, ep)),
+				res.CombineOutput().Bytes(),
+				0777)
+
+			if err != nil {
+				c.logCxt.Errorf("Can not create test results for %s err='%s'", command, err)
+			}
+		}
+	}
+
+	eps, err = c.GetEndpointsIdentityIds()
+	if err != nil {
+		c.logCxt.Errorf("Can not get endpoint identity ids")
+	}
+	for _, ep := range eps {
+		res := c.Node.Exec(fmt.Sprintf("sudo cilium identity get %s", ep))
+		err = ioutil.WriteFile(
+			fmt.Sprintf("%s/%s", testPath, fmt.Sprintf("identity_%s.txt", ep)),
+			res.CombineOutput().Bytes(),
+			0777)
+		if err != nil {
+			c.logCxt.Errorf("Can not create test results for identity get err='%s'", err)
+		}
+	}
+}
+
+//ReportEnd Write to the testReport directory all logs related with the Vagrant vm
+func (c *Cilium) ReportEnd() {
+	reportCommands := map[string]string{
+		"sudo journalctl -xe -u cilium --no-pager":           "cilium.log",
+		"sudo journalctl -xe -u cilium-docker --no-pager":    "cilium-docker.log",
+		"sudo docker logs cilium-consul":                     "consul.log",
+		`sudo bash -c "gops memstats $(pgrep cilium-agent)"`: "gops_memstats.txt",
+		`sudo bash -c "gops stack $(pgrep cilium-agent)"`:    "gops_stack.txt",
+		`sudo bash -c "gops stats $(pgrep cilium-agent)"`:    "gops_stats.txt",
+	}
+
+	gatherCommands := []string{
+		fmt.Sprintf("sudo cp -r /var/lib/cilium /vagrant/%s/lib", testResultsPath),
```

Comment:
Do not hardcode filepaths. Use constants that can be reused throughout the code. This will allow for easier refactoring if needed.